### PR TITLE
ENH docblock types

### PR DESCRIPTION
**Pull request by lekoala**
## Description
Fix/improve return types for public facing apis

## Manual testing steps
Nothing, just phpstan being happy

## Issues
- https://github.com/silverstripe/silverstripe-framework/issues/11173
- https://github.com/silverstripe/silverstripe-framework/issues/11149

## Pull request checklist

- [x] The target branch is correct
    - See [picking the right version](https://docs.silverstripe.org/en/contributing/code/#picking-the-right-version)
- [x] All commits are relevant to the purpose of the PR (e.g. no debug statements, unrelated refactoring, or arbitrary linting)
    - Small amounts of additional linting are usually okay, but if it makes it hard to concentrate on the relevant changes, ask for the unrelated changes to be reverted, and submitted as a separate PR.
- [x] The commit messages follow our [commit message guidelines](https://docs.silverstripe.org/en/contributing/code/#commit-messages)
- [x] The PR follows our [contribution guidelines](https://docs.silverstripe.org/en/contributing/code/)
- [x] Code changes follow our [coding conventions](https://docs.silverstripe.org/en/contributing/coding_conventions/)
- [x] This change is covered with tests (or tests aren't necessary for this change)
- [x] Any relevant User Help/Developer documentation is updated; for impactful changes, information is added to the changelog for the intended release
- [x] CI is green

### CODE REVIEW

File: src/Control/HTTPRequest.php [hunk 1]
```diff
@@ -463,9 +463,9 @@ public function offsetUnset(mixed $offset): void
      * <code>$this->setResponse(HTTPRequest::send_file('the content', 'filename.txt'));</code>
      *
      * @static
-     * @param $fileData
-     * @param $fileName
-     * @param null $mimeType
+     * @param string $fileData
+     * @param string $fileName
+     * @param string $mimeType
```
Comment (lozcalver) [inline src/Control/HTTPRequest.php]:
```suggestion
     * @param string|null $mimeType
```

**Review by lozcalver — CHANGES_REQUESTED** [review]
One minor tweak, otherwise good to merge

**Comment by GuySartorelli** [comment]
Can you please tick all of the appropriate checkboxes? It's intended both as a "have I done everything I need to?" checklist for you, as well as a "Is the contributor finished working on this or is it clear there is still more to do?" list for maintainers.
As it stands, with none of the boxes ticked, it looks like this isn't even ready to be reviewed.

**Comment by GuySartorelli** [comment]
Please also link to an **open** issue - we can't track closed issues

**Comment by lekoala** [comment]
@GuySartorelli  issue created https://github.com/silverstripe/silverstripe-framework/issues/11173

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
LGTM, it's an all 'round incremental improvement with no discernable side effects.

Please change the commit message from `FIX` to either `MNT` or `ENH` - `FIX` is for bugs that affect code, but this doesn't affect any of the actual code.

**Comment by GuySartorelli** [comment]
Ahh, I didn't notice this has a bunch of commits, so I'll have to squash merge anyway, so I can change the commit message during that process.

**Pull request merged**